src/array_string.rs

```diff
@@ -241,6 +241,68 @@ impl<const CAP: usize> ArrayString<CAP>
         }
     }
 
+    /// Adds the given `ArrayString` to the end of this `ArrayString`.
+    ///
+    /// ***Panics*** if the backing array is not large enough to fit the additional `ArrayString`.
+    ///
+    /// ```
+    /// use std::str::FromStr;
+    /// use arrayvec::ArrayString;
+    ///
+    /// let mut string = ArrayString::<2>::new();
+    /// let a = ArrayString::<1>::from_str("a").unwrap();
+    /// let b = ArrayString::<1>::from_str("b").unwrap();
+    ///
+    /// string.push_array_string(a);
+    /// string.push_array_string(b);
+    ///
+    /// assert_eq!(&string[..], "ab");
+    /// ```
+    #[track_caller]
+    pub fn push_array_string<const OTHER_CAP: usize>(&mut self, other: ArrayString<OTHER_CAP>) {
```

Comment:
The user should just use `string.push_str(&a)` here. Is there a reason not to do that?

Reply:
Ah would this work? I actually didn't know that! I assume this would also avoid heap allocs?

Reply:
of course, no heap allocations extra. ArrayString is just like another kind of String - they both have zero cost deref to &str.

Reply:
Ah perfect this can likely get closed out then!

Reply:
Ok, feel free to add an example that shows how this works, if you want to.